posthog-core/src/index.ts

```diff
@@ -268,9 +280,9 @@ export abstract class PostHogCore {
 
   groups(groups: { [type: string]: string | number }): this {
     // Get persisted groups
-    const existingGroups = this.props.$groups || {}
+    const existingGroups =
```

Comment:
also cc: @benjackwhite , the `reInit` was clearing persisted props, but this was sneaky, just using `this.props` directly.

Have replaced it in 3 places. Don't think it should affect anything else, and tests all pass, but have a look if you think it will?

Reply:
hmm, it's a bit confusing, I'm not quite sure why `this.props` exists. Can do away with it..

Reply:
Ah, I see, its the lazy loading getter which creates problems, as we can't tell when its been invalidated behind us.